homeassistant/components/overkiz/climate_entities/somfy_thermostat.py

```diff
@@ -0,0 +1,191 @@
+"""Support for TaHoma Smart Thermostat."""
+from __future__ import annotations
+
+from typing import Any, cast
+
+from pyoverkiz.enums import OverkizState
+
+from homeassistant.components.climate import ClimateEntity
+from homeassistant.components.climate.const import (
+    CURRENT_HVAC_HEAT,
+    CURRENT_HVAC_IDLE,
+    HVAC_MODE_AUTO,
+    HVAC_MODE_HEAT,
+    PRESET_AWAY,
+    PRESET_HOME,
+    PRESET_NONE,
+    SUPPORT_PRESET_MODE,
+    SUPPORT_TARGET_TEMPERATURE,
+)
+from homeassistant.const import ATTR_TEMPERATURE, TEMP_CELSIUS
+
+from ..coordinator import OverkizDataUpdateCoordinator
+from ..entity import OverkizEntity
+
+COMMAND_EXIT_DEROGATION = "exitDerogation"
+COMMAND_REFRESH_STATE = "refreshState"
+COMMAND_SET_DEROGATION = "setDerogation"
+COMMAND_SET_MODE_TEMPERATURE = "setModeTemperature"
+
+CORE_DEROGATED_TARGET_TEMPERATURE_STATE = "core:DerogatedTargetTemperatureState"
+CORE_DEROGATION_ACTIVATION_STATE = "core:DerogationActivationState"
+
+PRESET_FREEZE = "Freeze"
+PRESET_NIGHT = "Night"
+
+ST_HEATING_MODE_STATE = "somfythermostat:HeatingModeState"
+ST_DEROGATION_HEATING_MODE_STATE = "somfythermostat:DerogationHeatingModeState"
+
+STATE_DEROGATION_FURTHER_NOTICE = "further_notice"
+STATE_DEROGATION_ACTIVE = "active"
+STATE_DEROGATION_INACTIVE = "inactive"
+STATE_PRESET_AT_HOME = "atHomeMode"
+STATE_PRESET_AWAY = "awayMode"
+STATE_PRESET_FREEZE = "freezeMode"
+STATE_PRESET_MANUAL = "manualMode"
+STATE_PRESET_SLEEPING_MODE = "sleepingMode"
+STATE_PRESET_SUDDEN_DROP_MODE = "suddenDropMode"
+
+MAP_HVAC_MODES = {
```

Comment:
Could you check the naming in AtlanticElectricalHeater and change it here as well? So to make clear which side is Overkiz and which side is HA. Lets try to standardize our implementations.

Reply:
Done with da71ca0004